### PR TITLE
Need domain param for BLS

### PR DESCRIPTION
**Pull request by jrhea**
Just needed the domain param exposed.

### CODE REVIEW

File: crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java [hunk 1]
```diff
@@ -46,8 +45,8 @@ private BLS12381() {}
    * @param message The message to sign, not null
    * @return The SignatureAndPublicKey, not null
    */
-  public static SignatureAndPublicKey sign(KeyPair keyPair, byte[] message) {
-    G2Point hashInGroup2 = hashFunction(message);
+  public static SignatureAndPublicKey sign(KeyPair keyPair, byte[] message, int domain) {
```
Comment (atoulme) [inline crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java]: add the new parameter to javadoc real quick?

File: crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java [hunk 2]
```diff
@@ -62,8 +61,8 @@ public static SignatureAndPublicKey sign(KeyPair keyPair, byte[] message) {
    * @param message The message to sign, not null
    * @return The SignatureAndPublicKey, not null
    */
-  public static SignatureAndPublicKey sign(KeyPair keyPair, Bytes message) {
-    return sign(keyPair, message.toArray());
+  public static SignatureAndPublicKey sign(KeyPair keyPair, Bytes message, int domain) {
```
Comment (atoulme) [inline crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java]: same add parameter?

File: crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java [hunk 3]
```diff
@@ -75,10 +74,10 @@ public static SignatureAndPublicKey sign(KeyPair keyPair, Bytes message) {
    * 
    * @return True if the verification is successful.
    */
-  public static boolean verify(PublicKey publicKey, Signature signature, byte[] message) {
+  public static boolean verify(PublicKey publicKey, Signature signature, byte[] message, int domain) {
```
Comment (atoulme) [inline crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java]: same

File: crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java [hunk 4]
```diff
@@ -94,8 +93,8 @@ public static boolean verify(PublicKey publicKey, Signature signature, byte[] me
    * 
    * @return True if the verification is successful.
    */
-  public static boolean verify(PublicKey publicKey, Signature signature, Bytes message) {
-    return verify(publicKey, signature, message.toArray());
+  public static boolean verify(PublicKey publicKey, Signature signature, Bytes message, int domain) {
```
Comment (atoulme) [inline crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java]: same

File: crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java [hunk 5]
```diff
@@ -106,8 +105,8 @@ public static boolean verify(PublicKey publicKey, Signature signature, Bytes mes
    * 
    * @return True if the verification is successful, not null
    */
-  public static boolean verify(SignatureAndPublicKey sigAndPubKey, byte[] message) {
-    return verify(sigAndPubKey.publicKey(), sigAndPubKey.signature(), message);
+  public static boolean verify(SignatureAndPublicKey sigAndPubKey, byte[] message, int domain) {
```
Comment (atoulme) [inline crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java]: same

File: crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java [hunk 6]
```diff
@@ -118,12 +117,12 @@ public static boolean verify(SignatureAndPublicKey sigAndPubKey, byte[] message)
    * 
    * @return True if the verification is successful.
    */
-  public static boolean verify(SignatureAndPublicKey sigAndPubKey, Bytes message) {
-    return verify(sigAndPubKey.publicKey(), sigAndPubKey.signature(), message);
+  public static boolean verify(SignatureAndPublicKey sigAndPubKey, Bytes message, int domain) {
```
Comment (atoulme) [inline crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java]: last one

File: crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java (hunk 1 above)
Comment (cleishm) [reply crypto/src/main/java/net/consensys/cava/crypto/mikuli/BLS12381.java]: It's a "domain value", but an integer? Is there a range for this value? Or is it only certain discrete values? If the former, the range should probably be documented and asserted. If it's the later, then it should probably be an enum.

**Comment by jrhea** [comment]
> It's a "domain value", but an integer? Is there a range for this value? Or is it only certain discrete values? If the former, the range should probably be documented and asserted. If it's the later, then it should probably be an enum.

I'm experimenting with the math.  There is a fundamental issue with milagro that is making it difficult...hence, my rushed changes.  I wanted to use an experimental branch so that I didn't have to scrutinize every change.  instead i will just modify locally and stop pushing

**Comment by cleishm** [comment]
Is the issue access to a snapshot build for your experimental branch? I don’t see any reason why CI on this repo couldn’t be configured to push snapshots to bintray for an arbitrary branch, or perhaps even for all PR builds? Then you could experiment to your hearts content, and still get builds to try. The catch, of course, is how and when to conclude experiments and wrap them in. Cava modules, as a general toolkit, should of course be reasonably polished when they are published in release builds.

**Comment by jrhea** [comment]
> Is the issue access to a snapshot build for your experimental branch? I don’t see any reason why CI on this repo couldn’t be configured to push snapshots to bintray for an arbitrary branch, or perhaps even for all PR builds? Then you could experiment to your hearts content, and still get builds to try. The catch, of course, is how and when to conclude experiments and wrap them in. Cava modules, as a general toolkit, should of course be reasonably polished when they are published in release builds.

Totally agree that things should be polished.  That being said, I don't want to waste time polishing a 💩.

**Comment by cleishm** [comment]
Right. Hence an experiments branch? Although I guess you could also just do a `gradle install` and have it available locally anyway 🤷‍♀️

**Pull request merged**